### DISCUSSION TITLE
How to display line numbers in the page margins?

### DISCUSSION THREAD

**Discussion by grantmcdermott** [typst/typst#4098] [Q&A] [answered]
What it says in the title. Line numbering is commonly used for reviewing documents and for responding to referee requests. 

Basically, I'm looking for the Typst equivalent of

```{tex}
\usepackage(lineno}
\linenumbers
```

that I would use in LaTeX.

Apologies if I missed an existing and obvious solution. My search efforts didn't turn anything up....

**Comment by Enivex** [comment, marked as answer]
Not possible yet

https://github.com/typst/typst/issues/352

**Reply by agnel** [reply]
Oh... good, i chose not to switch from LaTeX to typst, _yet_.

Thanks for the information. But still looking forward to it.

**Reply by VibrantTreeStump** [reply]
@agnel The ticket was closed over a year ago. The feature has been implemented in August 2024 and has been available since Typst 0.12.0

**Reply by agnel** [reply]
> @agnel The ticket was closed over a year ago. The feature has been implemented in August 2024 and has been available since Typst 0.12.0

That's great!

Will check it out.

Thanks for the update.

**Reply by VibrantTreeStump** [reply]
You can check for yourself in the future by clicking on the linked issue

**Comment by grantmcdermott** [comment]
Ah, bummer. Thanks for the quick confirmation, though.

**Comment by MooersLab** [comment]
I, too, am looking forward to this feature. I would be valuable for the first submission of a scientific manuscript for review.

**Reply by PgBiel** [reply]
This is coming soon in Typst v0.12.0! You can test it right now by downloading v0.12.0-rc1 (release candidate 1) and giving us feedback. You can download the CLI from the [Releases Page](https://github.com/typst/typst/releases/tag/v0.12.0-rc1), but you can also test it in the web app by switching the Typst version for a project (in project settings) to "0.12 Testing".

**Reply by MooersLab** [reply]
Thank you! I downloaded v0.12.0. It works!

**Comment by davidzgli** [comment]
I need this feature to submit my manuscript too.

**Reply by grantmcdermott** [reply]
it's available as of typst v0.12.0: https://typst.app/docs/reference/model/par/#definitions-line

**Reply by davidzgli** [reply]
Wonderful!  Are any instructions documented how to apply this feature to the quarto documents?  I guess this feature need to be enabled in the YAML header somehow.   Thank you!

**Reply by Enivex** [reply]
> Wonderful! Are any instructions documented how to apply this feature to the quarto documents? I guess this feature need to be enabled in the YAML header somehow. Thank you!

You would have to ask in the quarto forum

**Reply by grantmcdermott** [reply]
@davidzgli just include it in your Quarto YAML header, e.g.

```yaml
title: "Important findings"
date: last-modified
format:
  typst:
    include-in-header:
      - text: |
          #set par.line(numbering: "1")
```

**Reply by davidzgli** [reply]
It works.  Really appreciate it -- solving my big headache.

**Reply by davidzgli** [reply]
> > Wonderful! Are any instructions documented how to apply this feature to the quarto documents? I guess this feature need to be enabled in the YAML header somehow. Thank you!
> 
> You would have to ask in the quarto forum

Sorry that I somehow came across this post.  Got lucky that my problem was solved here.
